synapse/handlers/room_member.py

```diff
@@ -908,6 +908,11 @@ async def _should_perform_remote_join(
                 Membership.JOIN,
                 Membership.INVITE,
             ):
+                # The event content should *not* include the authorising user as
+                # it won't be properly signed. Strip it out since it might come
+                # back from a client updating a display name / avatar.
+                content.pop(EventContentFields.AUTHORISING_USER, None)
```

Comment:
I debated about universally stripping this field, but that seemed like not a great solution.

Reply:
This feels like an odd side-effect for this method. Did you consider doing it in `update_membership_locked` or somesuch?

Reply:
I did consider doing it unconditionally near where we mess with the profile information. We don't have the room version that early / know if the user is in the room yet, so it would be a bit of a larger change.

Note that this method does already modify the `content` in some cases (by adding the `join_authorised_via_users_server` if it is needed), so it seemed somewhat reasonable that it also removes it in some cases?

I'd be OK changing that though if you think it is clearer!

Reply:
> Note that this method does already modify the `content` in some cases (by adding the `join_authorised_via_users_server` if it is needed), so it seemed somewhat reasonable that it also removes it in some cases?

Yeah, I did see that. It feels like adding it where it's needed is more expected than removing it where it's not, but ymmv.

I think stripping it universally would be fine.

I don't have a hugely strong opinion, but do think it would be clearer if you moved it out. Up to you though.

Reply:
> I think stripping it universally would be fine.

That's fine with me.